packages/react/src/themes/teams/components/Status/statusStyles.ts

```diff
@@ -1,81 +1,64 @@
-import { pxToRem, SizeValue } from '../../../../lib'
-import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
 import { StatusProps } from '../../../../components/Status/Status'
 import { StatusVariables } from './statusVariables'
+import { ComponentSelectorsAndStyles } from 'src/themes/types'
+import { backportComponentStyle } from 'src/lib/resolveComponentRules'
 
-const getBackgroundColor = (state: string, variables: StatusVariables) => {
-  switch (state) {
-    case 'success':
-      return variables.successBackgroundColor
-    case 'info':
-      return variables.infoBackgroundColor
-    case 'warning':
-      return variables.warningBackgroundColor
-    case 'error':
-      return variables.errorBackgroundColor
-    case 'unknown':
-    default:
-      return variables.defaultBackgroundColor
-  }
-}
+const statusStyles: ComponentSelectorsAndStyles<StatusProps, StatusVariables> = v => ({
+  root: [
+    [
+      null,
+      {
+        display: 'inline-flex',
```

Comment:
Component parts now define an array of `[selector, style]` tuples.  When the selector object matches the props of the component, then the style is applied.

A `null` selector matches any props object so the styles are always applied.

```jsx
const styles = {
  root: [
    [null, [{ display: 'block' }],
    [{ size: 'small', fontSize: '0.9em' }],
  ],
}
```
Here are the styles that would be produced:
```jsx
<Status />
// => { display: 'block' }

<Status size='small' foo />
// => { display: 'block', fontSize: '0.9em' }
```

The `foo` prop above does not prevent the match from being made.  The Status still has a `size` prop of `small`, so it matches the second root style.